src/jit/earlyprop.cpp

```diff
@@ -228,9 +230,9 @@ void Compiler::optEarlyProp()
 //    tree           - The input tree node to be rewritten.
 //
 // Return Value:
-//    Return true iff "tree" is successfully rewritten.
+//    Return a new tree if the original tree was successfully rewritten.
```

Comment:
You might also want to add a note that, if the tree is rewritten, its containing tree is updated (that wasn't obvious to me, given that previously the tree was modified in-place).